### PR TITLE
x-trim in strings will result in joi trim(true) injection

### PR DESCRIPTION
**Pull request by j-d-carmichael**
As the title says.

This is a very simple catch that will automatically trim input strings meaning the domain layer does not have to check this maually.

The dev can opt out of this by adding x-dont-trim to the string attribute of an object

With this PR the following openapi spec:
```yaml
type: object
required:
  - name
properties:
  name:
    type: string
    minLength: 1
    x-dont-trim: true
```
Will result in the Joi validation looking like:

```typescript
Joi.object({'name':Joi.string().min(1).required()...
```

Without the x-dont-trim present the output will default to:
```typescript
Joi.object({'name':Joi.string().trim(true).min(1).required()...
```

_______________

In addition to auto-trim, this PR permits the dev to make use of any of the Joi string methods: https://joi.dev/api/?v=17.9.1#string

The dev does this with the pattern:
```yaml
type: object
required:
  - name
properties:
  name:
    type: string
    x-joi-string-<add method here>: true
```
Any of the string methods work, eg lowercase and email:
```yaml
type: object
required:
  - name
properties:
  name:
    type: string
    x-joi-string-lowercase: true
    x-joi-string-email: true
```
the above would result in (in combo with the trim):
```typescript
Joi.object({'name':Joi.string().trim(true).lowercase().email().required()...
```

### CODE REVIEW

**Comment by acrontum-sanchez** [comment]
Sounds good to me.
Wont "x-trim: false" be the same as "x-dont-trim: true" ?

File: src/lib/helpers/SwaggerUtils.ts [hunk 1]
```diff
@@ -59,16 +57,41 @@ class SwaggerUtils {
...
+          if (typeof joiParam === 'object') {
+            joiParam = JSON.stringify(joiParam);
+          }
+          validationText += `.${joiStringMethod}(${joiParam})`;
+        }
+      });
+
+      if (type === 'string') {
+        const autoTrim = nodegenRc.joi?.strings?.autoTrim;
+        if (['opt-out', 'always'].includes(autoTrim)) {
```
Comment (p-mcgowan) [inline src/lib/helpers/SwaggerUtils.ts:76]: this includes is not required since the check below covers all cases

**Review by p-mcgowan — APPROVED** [review]

**Review by p-mcgowan — CHANGES_REQUESTED** [review]
issues generating in a test project (discussion in another channel)

File: .npmignore [hunk 2]
```diff
@@ -3,9 +3,14 @@
...
+.github
 __tests__
-githooks
 coverage
+docs
+githooks
+test_openapi3.yml
+test_asyncapi.yml
 test_swagger.yml
-src
```
Comment (p-mcgowan) [inline .npmignore:10]: src should be excluded from the node module

Comment (p-mcgowan) [reply .npmignore:10]: alternatively you can simplify by adding files to package.json

```json
"files": [
    "README.md",
    "dist/"
]
```
Then only output files and the readme are included in the node_module

**Review by p-mcgowan — APPROVED** [review]

**Pull request merged**